2020-02-Spring/0_arrays_matrix/README.md

```diff
@@ -119,21 +125,43 @@ Go to [Top](#top)
 <!-- Don't remove -->
 <a name="s2"/>
 
-### 2. SOLUTION 2 TODO :bug:
-
-Source: TODO :bug:
-
-#### Naive/Simple Solution
-
-TODO :bug:
-
-#### Optimal Solution
-
-TODO :bug:
-
-#### Testing The Solutions OR Driver For Solution
-
-TODO :bug:
+### 2. SOLUTION 2
+
+Source: https://leetcode.com/problems/maximum-subarray/
+
+We will traverse over the array and keep track of 2 variables: Total maximum sum, and the current sum.
+We will look at each number and add it to the current sum, if the current sum becomes negative, we ignore the numbers that made it negative.
+Resetting the current to 0 is basically saying "Ignore everything we looked at previously because it's not getting us the greatest sum."
+So if we are looking at this array: `[-2,1,-3,4,-1,2,1,-5,4]`
+We will look at -2 and see we have a negative number, so set curr to 0. 
+Then we see that 1 is postive, set curr & max to 1. 
+Then -3 gives us a curr of -2, set curr to 0, but the max is still 1.
+Then 4 gives us a curr of 4, and update max to 4. Since our curr is now 4, we have ignored all the values before 4 (-2, 1, -3).
+
+#### Solution
```

Comment:
Need a solution source too

Reply:
Solution is in the other file

Reply:
> Need a solution source too

Not sure what you're asking here but here's my 2 cents:
* I don't think I've ever given credit attribution to a solution before, mostly because I would typically write them myself (LC especially) or use the solution in the CTCI book.
* There is a `Solution.java` in this commit

Reply:
Oh, I thought we needed one for some reason. Never mind then.